src/monodroid/jni/inmemory-assemblies.cc

```diff
@@ -0,0 +1,155 @@
+#include "inmemory-assemblies.hh"
+#include "globals.hh"
+
+#include <mono/metadata/appdomain.h>
+#include <mono/metadata/assembly.h>
+#include <mono/metadata/reflection.h>
+
+extern "C" {
+#include "java-interop-util.h"
+}
+
+namespace xamarin { namespace android { namespace internal {
+
+void
+InMemoryAssemblies::add_or_update_from_java (MonoDomain *domain, JNIEnv *env, jstring_array_wrapper &assemblies, jobjectArray assembliesBytes)
+{
+	if (assembliesBytes == nullptr)
```

Comment:
Is this an expected condition? In other words, are there scenarios when this function can be called with the array being null and still be considered a valid call? If the answer is "no", then I think it should be an assert.

Reply:
Yep, this is to support the cases where the designers doesn't want to supply in-memory data and rely on the old behavior (which would be current designer while I finish the feature in our branch).

Reply:
I can move the null check into the caller site if that makes more sense